src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -116,18 +131,20 @@ private String getAsyncResultEndpoint(String jobId) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
     logger.info("Getting workspace {} for {}", id, userReq.getEmail());
     Workspace workspace = workspaceService.getWorkspace(id, userReq);
-    WorkspaceCloudContext cloudContext = workspaceService.getCloudContext(id, userReq);
-
-    // Note projectId will be null here if no cloud context exists.
-    // TODO: this assumes a GoogleContext is the only cloud context on a workspace. This will
-    // eventually need to change.
-    GoogleContext googleContext = new GoogleContext().projectId(cloudContext.googleProjectId());
+    GcpContext gcpContext =
+        workspace
+            .getGcpCloudContext()
+            .map(g -> new GcpContext().projectId(g.getGcpProjectId()))
+            .orElse(null);
+
+    // Note projectId will be null here if no GCP cloud context exists.
+    // When we have another cloud context, we will need to do a similar retrieval for it.
```

Comment:
I'd be more tempted to factor out the cloud context endpoints into separate controllers.
In general, I'm content to leave that refactoring to when we are adding the features rather than trying to guess ahead of time.